modules/programs/alacritty/default.nix

```diff
@@ -0,0 +1,234 @@
+  ];
+  modList = [
+    "Command"
+    "Control"
+    "Shift"
+    "Alt"
+  ];
+  modeList = [
+    "~AppCursor"
+    "AppCursor"
+    "~AppKeypad"
+    "AppKeypad"
+  ];
+in {
+  options.programs.alacritty = with types; {
+    enable = mkEnableOption "Alacritty terminal";
+    window = {
+      dimensions = {
+        columns = mkOption {
+          type = int;
+          default = 80;
+          description = "Number of columns in the window";
+        };
+        lines = mkOption {
+          type = int;
+          default = 24;
+          description = "Number of lines in the window";
+        };
+      };
+      padding = {
+        x = mkOption {
+          type = int;
+          default = 2;
+          description = "Horizontal padding around the window, in pixels";
+        };
+        y = mkOption {
+          type = int;
+          default = 2;
+          description = "Vertical padding around the window, in pixels";
+        };
+      };
+      decorations = mkOption {
+        description = ''
+          Window decorations
+            - full: Borders and title bar
+            - none: Neither borders nor title bar
+        '';
+        type = enum [ "none" "full" ];
+        default = "full";
+      };
+    };
+    scrolling = {
+      history = mkOption {
+        type = int;
+        description = ''
+          Maximum number of lines in the scrollback buffer.
+          Specifying '0' will disable scrolling.
+        '';
+        default = 10000;
+      };
+      multiplier = mkOption {
+        type = int;
+        description = "Number of lines the viewport will move for every line scrolled.";
+        default = 3;
+      };
+      autoScroll = mkOption {
+        type = bool;
+        description = "Scroll to the bottom when new text is written to the terminal.";
+        default = false;
+      };
+    };
+    font = let
+      fontOption = {
+        family = mkOption {
+          type = string;
+          description = "Font family to use.";
+          default = "monospace";
+        };
+        style = mkOption {
+          type = nullOr (enum ["Regular" "Bold" "Italic"]);
+          default = null;
+          description = "Font style to use.";
+        };
+      };
+    in {
+      normal = fontOption;
+      bold = fontOption;
+      italic = fontOption;
+      size = mkOption {
+        type = float;
+        description = "Point size.";
+        default = 11;
+      };
+      offset = {
+        x = mkOption {
+          default = 0;
+          type = int;
+          description = "Letter spacing.";
+        };
+        y = mkOption {
+          default = 0;
+          type = int;
+          description = "Line spacing.";
+        };
+      };
+      glyphOffset = {
+        x = mkOption {
+          default = 0;
+          type = int;
+          description = "Glyph offset determines the locations of the glyphs within their cells with the default being at the bottom. Increasing x moves it towards the right.";
```

Comment:
Best to avoid such long lines. Can have something like
```
          description = ''
            Glyph offset determines the locations of the glyphs within their
            cells with the default being at the bottom. Increasing x moves it
            towards the right.
          '';
```